src/network/serverpackethandler.cpp

```diff
@@ -1150,9 +1150,10 @@ void Server::handleCommand_Interact(NetworkPacket *pkt)
 			if (pointed_object->isGone())
 				return;
 
-			ItemStack punchitem = playersao->getWieldedItem();
+			ItemStack selected_item, hand_item;
+			ItemStack tool_item = playersao->getWieldedItem(&selected_item, &hand_item);
```

Comment:
`hand_item` isn't needed here, is it?

Reply:
It's needed as a parameter for the function so that it knows from which set it should take the non-empty stack. If "hand_item" is provided, it will fall back to the hand as soon the selected item is an empty ItemStack.